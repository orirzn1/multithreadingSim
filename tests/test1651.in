L9
S6
O10
N15

T0
I@0x00000000
SUB $5, $5, $2
LOAD $5, $2, 0x2
SUBI $6, $5, 0x4
STORE $1, $7, 0x1
STORE $3, $7, 0x4
SUB $3, $2, $3
SUBI $5, $3, 0x2
ADD $6, $2, $5
STORE $5, $6, 0x2
SUB $4, $6, $3
HALT $0

T1
I@0x00000000
SUB $4, $3, $5
SUBI $5, $7, 0x2
STORE $1, $5, 0x0
ADD $5, $6, $2
ADDI $0, $3, 0x4
LOAD $5, $0, 0x2
ADDI $4, $2, 0x4
HALT $0

T2
I@0x00000000
SUBI $5, $0, 0x1
ADD $6, $2, $6
ADD $5, $2, $3
LOAD $4, $7, $2
ADDI $4, $3, 0x3
HALT $0

T3
I@0x00000000
ADDI $5, $1, 0x2
ADDI $2, $5, 0x4
ADDI $4, $0, 0x1
SUBI $4, $7, 0x3
ADDI $2, $2, 0x3
LOAD $6, $2, 0x1
LOAD $1, $6, $7
STORE $0, $5, $6
HALT $0

T4
I@0x00000000
ADDI $3, $0, 0x2
ADD $3, $3, $1
STORE $4, $0, $3
SUBI $3, $1, 0x1
ADD $7, $0, $3
LOAD $1, $6, 0x1
ADD $1, $4, $4
ADDI $7, $6, 0x3
HALT $0

T5
I@0x00000000
ADD $0, $2, $1
SUBI $7, $7, 0x4
LOAD $3, $6, 0x4
ADD $7, $3, $6
HALT $0

T6
I@0x00000000
SUBI $0, $6, 0x4
STORE $3, $2, 0x2
ADDI $2, $3, 0x2
LOAD $1, $7, $2
ADD $1, $6, $2
SUB $7, $6, $0
ADD $3, $7, $2
LOAD $4, $2, $3
HALT $0

T7
I@0x00000000
LOAD $4, $6, 0x3
ADDI $3, $3, 0x3
SUBI $6, $0, 0x2
SUBI $2, $0, 0x4
ADD $0, $7, $6
SUB $0, $0, $4
SUB $7, $1, $5
STORE $2, $7, 0x3
ADD $0, $6, $6
HALT $0

T8
I@0x00000000
STORE $0, $3, $6
ADDI $7, $0, 0x3
SUB $6, $2, $1
SUB $2, $6, $0
LOAD $3, $5, 0x0
SUBI $3, $1, 0x2
SUBI $0, $2, 0x2
HALT $0

T9
I@0x00000000
STORE $5, $1, $7
SUB $1, $1, $3
ADD $7, $5, $3
SUB $3, $7, $4
SUBI $7, $0, 0x4
SUBI $1, $0, 0x2
LOAD $5, $0, $3
HALT $0

T10
I@0x00000000
STORE $4, $2, $2
ADD $3, $3, $2
ADDI $4, $1, 0x2
ADD $0, $5, $2
STORE $2, $7, $2
SUB $0, $5, $2
ADDI $2, $7, 0x1
SUBI $2, $4, 0x1
LOAD $7, $7, 0x0
STORE $5, $7, 0x1
HALT $0

T11
I@0x00000000
ADD $2, $7, $4
LOAD $2, $0, 0x2
STORE $1, $7, 0x3
SUB $1, $2, $6
SUBI $5, $6, 0x4
SUB $5, $0, $4
SUBI $7, $1, 0x2
ADD $6, $3, $6
HALT $0

T12
I@0x00000000
SUBI $0, $4, 0x3
ADD $4, $5, $4
ADD $4, $1, $0
SUBI $3, $5, 0x1
LOAD $2, $7, 0x3
LOAD $5, $5, 0x1
ADD $7, $3, $3
SUBI $6, $1, 0x2
HALT $0

T13
I@0x00000000
LOAD $3, $4, 0x3
SUB $3, $7, $4
STORE $1, $3, $6
STORE $4, $0, 0x0
LOAD $2, $7, 0x4
STORE $2, $5, 0x0
ADDI $3, $4, 0x1
HALT $0

T14
I@0x00000000
LOAD $3, $7, 0x2
STORE $5, $0, 0x0
ADD $0, $3, $1
SUB $6, $1, $6
SUB $6, $5, $2
SUBI $6, $6, 0x3
SUBI $7, $7, 0x2
SUB $6, $0, $7
LOAD $5, $3, 0x0
HALT $0

D@0x0
0x6
0x5
0x5
0x4
0x8
0x5
0x1
0x8
0x4
