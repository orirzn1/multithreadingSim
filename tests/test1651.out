
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFE	R5 = 0xFFFFFFFE	R6 = 0xFFFFFFFE	R7 = 0x0
Register file thread id 1:
	R0 = 0x4	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x4	R5 = 0x5	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x3	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 3:
	R0 = 0x0	R1 = 0x5	R2 = 0x9	R3 = 0x0	R4 = 0xFFFFFFFD	R5 = 0x2	R6 = 0x5	R7 = 0x0
Register file thread id 4:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0xFFFFFFFF	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x3
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x5	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x5
Register file thread id 6:
	R0 = 0xFFFFFFFC	R1 = 0x2	R2 = 0x2	R3 = 0x6	R4 = 0x5	R5 = 0x0	R6 = 0x0	R7 = 0x4
Register file thread id 7:
	R0 = 0xFFFFFFFC	R1 = 0x0	R2 = 0xFFFFFFFC	R3 = 0x3	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFE	R7 = 0x0
Register file thread id 8:
	R0 = 0xFFFFFFFE	R1 = 0x0	R2 = 0x0	R3 = 0xFFFFFFFE	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x3
Register file thread id 9:
	R0 = 0x0	R1 = 0xFFFFFFFE	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFC
Register file thread id 10:
	R0 = 0x0	R1 = 0x0	R2 = 0x1	R3 = 0x0	R4 = 0x2	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 11:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFE
Register file thread id 12:
	R0 = 0xFFFFFFFD	R1 = 0x0	R2 = 0x0	R3 = 0xFFFFFFFF	R4 = 0xFFFFFFFD	R5 = 0x0	R6 = 0xFFFFFFFE	R7 = 0xFFFFFFFE
Register file thread id 13:
	R0 = 0x0	R1 = 0x0	R2 = 0x2	R3 = 0x1	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 14:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x2	R7 = 0xFFFFFFFE
Blocked MT CPI for this program 4.969231

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFE	R5 = 0xFFFFFFFE	R6 = 0xFFFFFFFE	R7 = 0x0
Register file thread id 1:
	R0 = 0x4	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x4	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x3	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 3:
	R0 = 0x0	R1 = 0x0	R2 = 0x9	R3 = 0x0	R4 = 0xFFFFFFFD	R5 = 0x2	R6 = 0x5	R7 = 0x0
Register file thread id 4:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0xFFFFFFFF	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x3
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x2	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x2
Register file thread id 6:
	R0 = 0xFFFFFFFC	R1 = 0x2	R2 = 0x2	R3 = 0x6	R4 = 0x5	R5 = 0x0	R6 = 0x0	R7 = 0x4
Register file thread id 7:
	R0 = 0xFFFFFFFC	R1 = 0x0	R2 = 0xFFFFFFFC	R3 = 0x3	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFE	R7 = 0x0
Register file thread id 8:
	R0 = 0xFFFFFFFE	R1 = 0x0	R2 = 0x0	R3 = 0xFFFFFFFE	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x3
Register file thread id 9:
	R0 = 0x0	R1 = 0xFFFFFFFE	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFC
Register file thread id 10:
	R0 = 0x0	R1 = 0x0	R2 = 0x1	R3 = 0x0	R4 = 0x2	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFE
Register file thread id 11:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFE
Register file thread id 12:
	R0 = 0xFFFFFFFD	R1 = 0x0	R2 = 0x0	R3 = 0xFFFFFFFF	R4 = 0xFFFFFFFD	R5 = 0x0	R6 = 0xFFFFFFFE	R7 = 0xFFFFFFFE
Register file thread id 13:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x1	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 14:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFE	R6 = 0x2	R7 = 0xFFFFFFFE
Finegrained Multithreading CPI for this program 1.061538

